### ISSUE TITLE
Prepare plugin does not work for texture atlas

### ISSUE THREAD

**Issue by molst** [pixijs/pixi.js#6428]
Hi,
I just noticed that preparing individual objects in a texture atlas works, but preparing the whole atlas once and for all does not work. According to the plugin docs, prepare should accept a BaseTexture, but this does not give any performance boost in my case.

The 'actiondetails' texture referenced below is a 4096x4096 atlas.

``` js
    var hull = new PIXI.Sprite(textures.get('actiondetails')[`${scls.toFilename(newObj.id)}_hull.png`]);
    renderer.plugins.prepare.upload(hull); //SPEEDY on first GPU load!
    //renderer.plugins.prepare.upload(textures.get('actiondetails')); //SLOW on first GPU load!
```

### Environment

- **`pixi.js` version**: 5.2.0
- **`platform`**: Windows 10 desktop and iPhone 6

**Comment by ivanpopelyshev** [comment]
What prepare does: it goes recursively in scene and searches for BaseTexture's. 

Its not possible to upload single Texture of atlas, only whole atlas has be uploaded. It actually calls `renderer.texture.bind(myAtlasBaseTexture)`  - that's all that it does.

https://github.com/pixijs/pixi.js/blob/805387d190d36812fe53d382f3ddc5242d4d6a06/packages/prepare/src/Prepare.js#L68-L70

Prepare/BasePrepare classes are plain JS code , no webgl magic. People found many problems with it in v5, you should debug it for your case too :)

**Comment by molst** [comment]
Okidoke. I have located the problem.

First, there was an error on my side. I supplied a texture received from the resource loader, which is not of PIXI.BaseTexture type. So naturally, there was no hook for it in Prepare.

However, when fixing that error and supplying a BaseTexture as received from the resource loader, this didn't work either. The reason for this is that there is no hook for BaseTexture's, but only for objects **having** a baseTexture as a property, [look here](https://github.com/pixijs/pixi.js/blob/805387d190d36812fe53d382f3ddc5242d4d6a06/packages/prepare/src/BasePrepare.js#L358).

So, either I think [this doc](https://github.com/pixijs/pixi.js/blob/805387d190d36812fe53d382f3ddc5242d4d6a06/packages/prepare/src/BasePrepare.js#L124) should be clarified, or the hooks should be updated to support that the supplied item actually **is** of the documented types.
I suggest the latter solution.

With the current design, however, it works fine to supply the resource loader's `resource.spritesheet` member to the Prepare.upload method if immediate preparation of all loaded resources is intended.
